File: euler_012.py
from time import time
from primePy import primes
from collections import Counter

def triangle():
    n = 1
    while True:
        yield int(n*(n+1)/2)
        n += 1

tri_gen = triangle()

def number_of_factors(n):
    num = 1
    for i in Counter(primes.factors(n)).values():
        num *= i+1
    return num

start = time()
t = next(tri_gen)
while True:
    if number_of_factors(t) > 500:
        print("First triangle number with more than 500 divisors: " + str(t))
        break
    t = next(tri_gen)


stop = time()
print("Time: " + str(stop-start))

File: euler_012.js
const start = Date.now();

function noOfFactors(n) {
    if (n === 1) {return 1;}

    let factors = [1, n];
    let max = n;

    for (let i = 2; i < max; i++) {
        if (n % i === 0) {
            factors.push(i);

            let r = n / i;
            if (r !== i) {
                factors.push(r);
            }
            max = r;
        }
    }

    return factors.length;
}


let triangleNumber = 28;
let n = 7;

while (noOfFactors(triangleNumber) < 501) {
    n++;
    triangleNumber += n;
}

const end = Date.now();

console.log("Execute time: ", end - start, " First triangle number is: ", triangleNumber);